spring-cloud-stream-binder-rabbit-core/src/main/java/org/springframework/cloud/stream/binder/rabbit/properties/RabbitProducerProperties.java

```diff
@@ -202,6 +203,10 @@ public void setRoutingKeyExpression(Expression routingKeyExpression) {
 		this.routingKeyExpression = routingKeyExpression;
 	}
 
+	public void setRoutingKey(String routingKey) {
+		setRoutingKeyExpression(new LiteralExpression(routingKey));
+	}
+
```

Comment:
Unfortunately, this doesn't comply with the properties contract; it needs a corresponding getter and a field with a comment (for auto README doc generation).

Instead of creating a `LiteralExpression` here; it should be created in `getRoutingKeyExpression()` if the expression is null (and this is not).